CMakeLists.txt

```diff
@@ -3,6 +3,8 @@ cmake_minimum_required(VERSION 3.20)
 
 
 ############### Subdirectories for dependencies ###############
+set(CMAKE_CXX_FLAGS_RELEASE "/MT /O2 /Ob2 /DNDEBUG")
+set(CMAKE_CXX_FLAGS_DEBUG "/MT /Zi /Ob0 /Od /RTC1")
```

Comment:
Please see these linkage errors:
https://dev.azure.com/Necktschnagge/341439fd-7b7d-482a-8ae7-6ae03646a61d/_apis/build/builds/11007/logs/60

The problem is about using dynamic vs static runtime library on windows. There are some flags to tell the compiler what to use.
CMake creates multiple targets, especially our Tobor_static_lib target as well as the binaries of googletest resulting from`add_subdirectory(extern/googletest)`.
The Test-Tobor target created in subdirectory test creates the executable Test-Tobor.exe.
This application must not use libraries which use different versions of multithreaded runtime library.
See Microsofts explainations here:
https://learn.microsoft.com/de-de/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170

May you figure out how to fix this @r0ost3r ?
If you gain any knowledge, please write it down, e.g. what is the default behaviour, which target needs to be modified?

Reply:
Currently Azure cannot load a bunch of dependencies.
Better wait some time and hope it will be working again soon.

Reply:
Use this (since Cmake 3.15):
https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html

Reply:
I solved this issue.
using `cmake --verbose` the log shows us which targets used which flag (`/MTd` or `/MDd`):
**/MTd:** googletest(2 times googletest/googletest and 2 times googletest/googlemock)
**/MDd:** src/ (2 times, one for .exe, one for .lib) + test/

The solution is to [make googletest use MDd](https://stackoverflow.com/questions/12540970/how-to-make-gtest-build-mdd-instead-of-mtd-by-default-using-cmake)